File: android/app/src/main/java/com/myfabricapp/UgoiraViewModule.java
package com.myfabricapp;

import android.graphics.drawable.AnimationDrawable;
import android.os.Handler;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactModule;

@ReactModule(name = "UgoiraViewModule")
public class UgoiraViewModule extends ReactContextBaseJavaModule {
    private AnimationDrawable animationDrawable;
    private final Handler handler = new Handler();
    private boolean isPaused = false;

    public UgoiraViewModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "UgoiraViewModule";
    }

    @ReactMethod
    public void pauseAnimation(Promise promise) {
        if (animationDrawable == null) {
            promise.reject("ERROR", "Animation is not initialized.");
            return;
        }
        if (!isPaused) {
            isPaused = true;
            handler.post(new Runnable() {
                @Override
                public void run() {
                    animationDrawable.stop();
                    promise.resolve(null);
                }
            });
        } else {
            promise.reject("ERROR", "Animation is already paused.");
        }
    }

    @ReactMethod
    public void resumeAnimation(Promise promise) {
        if (animationDrawable == null) {
            promise.reject("ERROR", "Animation is not initialized.");
            return;
        }
        if (isPaused) {
            isPaused = false;
            handler.post(new Runnable() {
                @Override
                public void run() {
                    animationDrawable.start();
                    promise.resolve(null);
                }
            });
        } else {
            promise.reject("ERROR", "Animation is already running.");
        }
    }

    public void setAnimationDrawable(AnimationDrawable animationDrawable) {
        this.animationDrawable = animationDrawable;
    }
}

File: android/app/src/main/kotlin/com/myfabricapp/UgoiraViewModule.kt
package com.myfabricapp

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.modules.core.DeviceEventManagerModule
import android.os.Bundle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner

class UgoiraViewModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext), LifecycleObserver {

    private var isAnimationPaused: Boolean = false

    init {
        reactContext.addLifecycleEventListener(object : LifecycleOwner {
            override fun getLifecycle() = reactContext.lifecycle
        })
    }

    override fun getName(): String {
        return "UgoiraView"
    }

    @ReactMethod
    fun pauseAnimation(promise: Promise) {
        isAnimationPaused = true
        promise.resolve(null)
    }

    @ReactMethod
    fun resumeAnimation(promise: Promise) {
        isAnimationPaused = false
        promise.resolve(null)
    }

    @ReactMethod
    fun isAnimationPaused(promise: Promise) {
        promise.resolve(isAnimationPaused)
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy() {
        isAnimationPaused = false
        // Implement additional resource management here
    }
}